src/server/index.js

```diff
@@ -80,16 +96,26 @@ function Server({ games, db }) {
       });
 
       socket.on('sync', (gameID, playerID, numPlayers) => {
```

Comment:
I think we should use some sort of `userID` here instead of `playerID`. `playerID` should typically be '0', '1', '2' and so on (some of the turn order logic even relies on this, but will be made more generic eventually).

Each game has a fixed set of `playerID`'s that don't change. The model I have in mind is:
- `userID`'s can be associated with these `playerID`'s for a particular game in `createRoom`. This mapping can even be dynamic (users dropping in and out of games, or being replaced by bots).
- `userID` can be associated with a client similar to the way `playerID` currently is [here](https://github.com/google/boardgame.io/blob/master/src/client/client.js#L66). You probably need to send a secret along as well so that we can be sure that that client is indeed the `userID` it claims to be.
- Unauthenticated games can just send a `playerID` and things should work as before.

Reply:
This sounds good. Just got back from a trip so will dig more into these changes later this week.